store/iavlstore.go

```diff
@@ -42,14 +43,17 @@ type iavlStore struct {
 
 	// How many old versions we hold onto.
 	// A value of 0 means keep all history.
-	numHistory int64
+	numRecent int64
+
+	storeEvery int64
```

Comment:
Added comments, and changed semantics so that numRecent=0 and storeEvery=0 are meaningful and won't throw errors. With numRecent=0 we will only store the current state tree, and not store any history. With storeEvery=0 we will not store any waypoints and only store the current tree and whatever is stored by numRecent. Also the option to not prune at all is now enabled by storeEvery=1, rather than numHistory=0.